tests/Unit/Utilities/Test_Algorithm.cpp

```diff
@@ -0,0 +1,39 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <deque>
+#include <vector>
+
+#include "Utilities/Algorithm.hpp"
+
+SPECTRE_TEST_CASE("Unit.Utilities.indices_of", "[Unit][Utilities]") {
+  const auto helper = [](const auto& data) {
+    CHECK(std::array<size_t, 3>{{0, 1, 3}} ==
+          indices_of<3>(data, [](const double lhs, const double rhs) {
+            return lhs < rhs;
+          }));
+
+    CHECK(std::array<size_t, 3>{{0, 1, 2}} ==
+          indices_of<3>(data, [](const double lhs, const double rhs) {
+            return std::abs(lhs) < std::abs(rhs);
+          }));
+
+    CHECK(std::array<size_t, 3>{{4, 1, 2}} ==
+          indices_of<3>(data, [](const double lhs, const double rhs) {
+            return lhs > rhs;
+          }));
+
+    CHECK(std::array<size_t, 3>{{3, 4, 2}} ==
+          indices_of<3>(data, [](const double lhs, const double rhs) {
+            return std::abs(lhs) > std::abs(rhs);
+          }));
+  };
+  helper(std::array<double, 5>{{-1.0, 7.8, 9.3, -22.8, 9.5}});
+  helper(std::deque<double>{-1.0, 7.8, 9.3, -22.8, 9.5});
+  helper(std::vector<double>{-1.0, 7.8, 9.3, -22.8, 9.5});
```

Comment:
Test with DataVector too (`indices_of` is being added for use in #705, where the input is a DataVector).